### PR TITLE
Make it easier to change the framework for a stub brief

### PR DESCRIPTION
**Pull request by lfdebrux**
Also includes a few fixes to other stubs.

I wanted to make it easier to change the framework for a stub brief, but this ended up being a bit of a rabbit hole as I ran into an existing bug.

I'm happy to split it into multiple PRs if reviewers would prefer that.

### CODE REVIEW

File: dmtestutils/api_model_stubs/brief.py [hunk 1]
```diff
@@ -60,6 +60,15 @@ def __init__(self, **kwargs):
...
                 del self.response_data[snakecase_kwarg]
 
+        if (
+            ("framework_slug" in kwargs or "frameworkSlug" in kwargs)
+            and ("framework_name" not in kwargs and "frameworkName" not in kwargs)
+            and ("framework_family" not in kwargs and "frameworkFramework" not in kwargs)
+        ):
+            self.response_data.update(self._format_framework(self.response_data["frameworkSlug"], oldstyle=True))
+            self.response_data["frameworkFramework"] = self.response_data["frameworkFamily"]
+            del self.response_data["frameworkFamily"]
```
[the remark this answers is not in the document]
Comment (lfdebrux) [reply dmtestutils/api_model_stubs/brief.py:70]: I'm not sure I understand the deprecation plan, would you be happy to add the comment? 😅

[the remark this answers is not in the document]
Comment (katstevens) [reply dmtestutils/api_model_stubs/brief.py:70]: Well the plan itself stalled somewhat, hence different models having different combinations of fields. Will see if I can do one of these fancy Github suggestions...

Comment (katstevens) [inline dmtestutils/api_model_stubs/brief.py:70]:
```suggestion
            # The Brief model serialisation still includes the deprecated `frameworkFramework` field. 
            # Use `brief['framework']['family']`, not `brief['frameworkFamily']` or `brief['frameworkFramework]`. 
            del self.response_data["frameworkFamily"]
```

Comment (lfdebrux) [reply dmtestutils/api_model_stubs/brief.py:72]: Thank you 😁

**Pull request merged**